doc/ubuntu_release_process.md

```diff
@@ -115,42 +124,55 @@ Last, we need to push our tag above.
 
 
 ### cherry-picked changes ###
-This is generally not the mechanism that is preferred for any release supported by trunk.  It may be used in order to get a upload in *really quickly* for a hot fix.
+This is generally not the preferred mechanism for any release supported by trunk.  It is used either to get a upload in *really quickly* for a hot fix or during release **Feature Freeze** when only bugfixes or FeatureFreezeExceptions (FFEs) are accepted.  The tool for doing this is in ``uss-tableflip/scripts/cherry-pick``.  It takes as import a commit-ish that it will create a patch to apply the commitish via debian/patches/cpick-\*.
 
 #### Cherry-pick Process
-The tool for doing this is in ``uss-tableflip/scripts/cherry-pick``.  It takes as import a commit-ish that it will create a cherry-pick from.
-
-    ## just tag current so you have a revert point.
-    $ git tag --delete xx-current >/dev/null 2>&1; git tag xx-current
-    $ git checkout ubuntu/xenial
-    $ cherry-pick dc2bd79949
-
-The tool will:
-
-  * add a new file to debian/patches/ named cpick-<commit>-topic that is dep-8 compatible
-  * add that patch to debian/patches/series
-  * verify that the patch applies (quilt push -a)
-  * give you a chance to edit formatted the debian/changelog entry
-  * prompt you to commit the change.
+```
+  $ git fetch upstream
+  $ git checkout -b ubuntu/xenial upstream/ubuntu/xenial || git checkout ubuntu/xenial
```

Comment:
`cherry-pick` tool does that for us automatically. it'll checkout ubuntu/daily/devel and wipe that out with git reset --hard ubuntu/devel latest commit (which includes the latest cherry-pick that was just performed).

I'm not as worried about documenting --hard wiping ubuntu/daily/* branches because we never expect to have someone working on those branches due to the fact that they are only around to enable dropping cherry-picks.

Reply:
> we never expect to have someone working on those branches due to the fact that they are only around to enable dropping cherry-picks.

I'm not sure I share this opinion. There are other ways in which the daily branch could be used. One time we could have used the daily branch in another way, for example, is when we migrated from nose to pytest. As it was, we had to have both test frameworks in the Build-Depends of `ubuntu/*` so that daily builds (with pytest) would pass, and archive builds (with nose) would pass. With a separate daily branch, we could have only made the Build-Depends change there (and we could have removed nose from the Build-Depends). This would have kept our archive builds more accurate because the wouldn't have a needless pytest dependency. It would also have given us more confidence that the archive builds would continue to be successful after the change of test framework, because the daily builds would be testing building with _only_ pytest (so any inadvertent dependency on nose would cause build failures).

(This is why I'm still unconvinced that writing tooling to fully automate this is the right thing to do: it forces us to consider all possible cases up front, which is bound to fail.)

Reply:
> (This is why I'm still unconvinced that writing tooling to fully automate this is the right thing to do: it forces us to consider all possible cases up front, which is bound to fail.)

Addressing our current needs does not need to be an exhaustive effort of automating all possible cases up front; it is sufficient to automate the ones we currently have.

Forcing a consideration of additional cases is useful and we are not required just by knowing of additional scenarios where cherry-pick will not work to actually implement changes for those scenarios.

I'm much happier having a single, if not "fully automated", tool to address our current issues with the daily recipe building than just a documented process with commands which humans may forget a step or three.

Reply:
> I'm not sure I share this opinion. There are other ways in which the daily branch could be used. One time we could have used the daily branch in another way, for example, is when we migrated from nose to pytest. 

@OddBloke Your thought is correct here. We could have carried a one-off patch in ubuntu/daily/$release that handled the proper build-deps just so our daily builds succeed. 

Yet, I think that the scenario you mention involved the alignment of two fairly infrequent occurrences:

 1. most ubuntu/$releases don't sit with active cherry-picks for a long time before exiting with a new-upstream-snapshot due to a subsequent SRU
 2. we don't generally carry inconsistencies for our daily recipe builders as we saw with the shift to pytest

It doesn't preclude this from happening again, but I think we can handle the 90% typical case here with automation, document the behavior of the scripts and if we run into special cases like daily build support that is unique during a cherry-pick phase of a release, then we manually update ubuntu/daily/$release ourselves instead of using the automation.

> (This is why I'm still unconvinced that writing tooling to fully automate this is the right thing to do: it forces us to consider all possible cases up front, which is bound to fail.)

I agree that productizing this release script to fully automate branching strategies for all use-cases is *not* the right thing to do, but we can easily handle to typical case for our cherry-picks and daily recipe builds and if we find ourselves having to  manually override the typical use-case multiple times, then lets find a better way to automated it, or drop the automation as the expected typical case doesn't seem so typical anymore.